### PR TITLE
feat(calling): Display others video in an ongoing call (AR-1239)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1239" title="AR-1239" target="_blank">StoryAR-1239</a>  1:1 video call
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

This PR aims to display the video feed of all call participants that have a camera on

### Dependencies (Optional)

Must be released with :
[feat(calling): receive video stream (AR-1232)
](https://github.com/wireapp/kalium/pull/793)

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

Start a call, if other users enable their video you should see them on the grid

### Notes (Optional)

Next PR we will handle ordering, users with video should be taken to the top 

### Attachments (Optional)

Wire 2022-08-24 at 5_33 PM

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-860/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4238892)(4238892)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4238892)(4238892)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-860/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4244693)(4244693)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4244693)(4244693)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/calling/common/CallingGridView.kt [hunk 1]
```diff
@@ -33,46 +39,63 @@ fun GroupCallGrid(
...
+        contentPadding = PaddingValues(MaterialTheme.wireDimensions.spacing4x),
         horizontalArrangement = Arrangement.spacedBy(MaterialTheme.wireDimensions.spacing2x),
         verticalArrangement = Arrangement.spacedBy(MaterialTheme.wireDimensions.spacing2x),
         columns = GridCells.Fixed(NUMBER_OF_GRID_CELLS)
     ) {
 
-        items(participants) { participant ->
+        items(items = participants, key = { it.id.toString() }) { participant ->
+            // since we are getting participants by chunk of 8 items,
+            // we need to check that we are on first page for sel user
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/calling/common/CallingGridView.kt:50]:
```suggestion
            // we need to check that we are on first page for self user
```

File: app/src/main/kotlin/com/wire/android/ui/calling/common/OneOnOneCallView.kt [hunk 2]
```diff
@@ -23,33 +32,56 @@ fun OneOnOneCallView(
...
     ) {
-        participants.forEach { participant ->
-            // For now we are handling only self user camera state
-            val isCameraOn = if (pageIndex == 0 && participants.first() == participant)
-                isSelfUserCameraOn else false
-            val isMuted = if (pageIndex == 0 && participants.first() == participant) isSelfUserMuted
-                else participant.isMuted
+        items(items = participants, key = { it.id.toString() }) { participant ->
+            // since we are getting participants by chunk of 8 items,
+            // we need to check that we are on first page for sel user
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/calling/common/OneOnOneCallView.kt:43]:
```suggestion
            // we need to check that we are on first page for self user
```

**Review by alexandreferris — APPROVED** [review]
👌🏻

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-860/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4255517)(4255517)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4255517)(4255517)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-860/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4255896)(4255896)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4255896)(4255896)-dev-release.apk)

**Pull request merged**